### ISSUE TITLE
Intermittent UI test failure - SwitchLocaleTest.FrenchLocaleTest

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/focus-android#4851]
### Firebase Test Run:
Failing 2/2 tries on FTL and locally:
https://console.firebase.google.com/u/1/project/moz-fx-mobile-firebase-testlab/testlab/histories/bh.2b4ca2c7fdb2f0f0/matrices/5461399497652579435/executions/bs.3b8fc141094e148d/testcases/9
### Stacktrace:
ndroidx.test.uiautomator.UiObjectNotFoundException: UiSelector[TEXT=Général, RESOURCE_ID=android:id/title]
at androidx.test.uiautomator.UiObject.click(UiObject.java:416)
at org.mozilla.focus.activity.SwitchLocaleTest.FrenchLocaleTest(SwitchLocaleTest.kt:217)
### Build: 
Debug 5/3
Started failing with this [commit](https://github.com/mozilla-mobile/focus-android/commit/c4f580942d7f81104a1d4e6be3934e92d8f5f591).

**Comment by sv-ohorvath** [comment]
CC-ing @pocmo to have a look here, because I tried to reproduce the issue manually to find the bug here but I can't figure out what it is. 
The test only started failing with this [commit](https://github.com/mozilla-mobile/focus-android/commit/c4f580942d7f81104a1d4e6be3934e92d8f5f591) so I'm only guessing that might be causing something I can't see.
The test also fails locally if you run both locale tests in this class, or this test twice. I tried to restart the app in the test, after changing the system locale but it's still not translating the app.
Testing manually it seems to work fine.
